benchmark/protocol/src/main/scala/Protocol.scala

```diff
@@ -59,7 +59,7 @@ object protocol {
       .codec[MediumW]
       .codec[SmallW]
       .codec[BigW]
-      .specify1[LargeW, LargeW]("identityLarge")
-      .specify1[MediumW, MediumW]("identityMedium")
-      .specify1[BigW, BigW]("identityBig")
+      .specify1("identityLarge", Field.strict[LargeW]("in"), Type[LargeW])
```

Comment:
@jedesah Perhaps some appropriate data structure / functionality from shapeless could help catching things at compile time and enforce requirements better :)